pytorch_lightning/profiler/profilers.py

```diff
@@ -401,14 +410,36 @@ def __init__(
 
         self.output_fname = output_filename
         self.output_file = None
+        self.local_rank = local_rank
+        if self.local_rank is not None:
+            self.on_fit_start(self.local_rank)
+            self.on_fit_start = super().on_fit_start
+
+    def on_train_start(self, local_rank: Optional[str] = None):
+        self.local_rank = local_rank
+
+        if local_rank != 0:
+            self.wrap_functions_into_rank_zero_only()
 
         if self.output_fname:
+            if local_rank is not None:
+                if '.txt' not in self.output_fname:
+                    raise MisconfigurationException("Log file should be .txt file.")
+            
+                self.output_fname = self.output_fname.replace(".txt", f"_{self.local_rank}.txt")
+
             fs = get_filesystem(self.output_fname)
             self.output_file = fs.open(self.output_fname, "w")
 
         streaming_out = [self.output_file.write] if self.output_file else [log.info]
         super().__init__(output_streams=streaming_out)
 
+    def wrap_functions_into_rank_zero_only(self):
+        self.start = rank_zero_only(self.start)
+        self.stop = rank_zero_only(self.stop)
+        self.summary = rank_zero_only(self.summary)
+        self.describe = rank_zero_only(self.describe)
+
```

Comment:
How comes wrapping these functions to run only on rank 0 works? Don't all the other processes need to run the start/stop logic for them to profile correctly?

Reply:
Good catch ! Missing condition

Reply:
```
        # when logging to `log.info`, only perform profiling on rank 0
        if local_rank != 0 and self.output_fname is None:
            self.wrap_functions_into_rank_zero_only()
```